### ISSUE TITLE
Does this module work?

### ISSUE THREAD

**Issue by mflaxman** [getsentry/raven-aiohttp#15]
The following simple script (`test.py`) seems to demonstrate that this module doesn't work?

```
import asyncio
from raven import Client
from raven_aiohttp import AioHttpTransport


async def function_with_error(client):
    try:
        1 / 0
    except:
        print(client.captureException())


if __name__ == '__main__':
    # client = Client(transport=AioHttpTransport)  # this fails silently!
    client = Client()  # this succeeds, but has obvious performance issues

    assert client.is_enabled()

    event_loop = asyncio.get_event_loop()
    event_loop.run_until_complete(function_with_error(client))
```

This simplest way to run this yourself is to `cp` this script to your `/tmp` directory and then run the following:
```
$ docker run -it -v "/tmp/:/tmp/" python:3.6 bash
```
From inside the container:
```
root@2f62958daa22:/# pip install aiohttp raven raven_aiohttp
root@2f62958daa22:/# SENTRY_DSN=https://<secret>:<key>@foo.com/2 python3 tmp/test.py
5eae13716f7b4988a4c5917e00719665
Sentry is attempting to send 1 pending error messages
Waiting up to 10 seconds
Press Ctrl-C to quit
```
Check your sentry logs, you should see an event with the id `5eae13716f7b4988a4c5917e00719665`. Success!

Now edit the `test.py` to see toggle the `AioHttpTransport` class and run it again inside the container:
```
root@2f62958daa22:/# SENTRY_DSN=https://<secret>:<key>@foo.com/2 python3 tmp/test.py
cf235077fadb42b6a7c84a28960436d1
```
The debug print goes away, but more importantly if you check your sentry logs, no event was received!

Sorry if I'm missing something obvious!

**Comment by mflaxman** [comment]
I found what it was.

I assumed that `run_until_complete` would run until it was complete, but it's a little more complicated. `async_send` is a synchronous method that schedules an asynchronous task. If the event loop is closed before that asynchronous task completes, then the synchronous method will respond with the event ID yet the task will not have been executed (meaning no event was sent to your sentry server).

You can see this more clearly if you add the following to the end of`test.py`:
`event_loop.close()`

If you have a long running process (like a web server), this is not a problem. Otherwise, it's actually quite tricky to fix. For example, in my test script I just add a `asyncio.sleep(1)` to the end of `function_with_error` and then it works. But, if you got 429'ed and have a lot of retries, then the caller won't know how long to sleep. I'm not sure if there's an easy way to clean that up.

**Comment by Kentzo** [comment]
AioHttpTransport needs a cleanup method that can be called to properly await all pending requests.

**Comment by hellysmile** [comment]
Here is a lot of proposed changes https://github.com/getsentry/raven-aiohttp/pull/18

**Comment by Kentzo** [comment]
I belive this issue can be closed because 0.6.0 was released.
